pkg/providers/v1/aws.go

```diff
@@ -1558,6 +1559,15 @@ func (c *Cloud) ensureSecurityGroup(ctx context.Context, name string, descriptio
 		createRequest.GroupName = &name
 		createRequest.Description = &description
 		tags := c.tagging.buildTags(ResourceLifecycleOwned, additionalTags)
+		if _, ok := tags["Name"]; !ok {
+			tags["Name"] = name
+		}
+		// TODO/TBD do we need to introduce a managed tag to ensure SG wont be leaked if config is changed when LB is created in managed mode?
```

Comment:
this is important, it could cause leak if we dont introduce a webhook or mechanism to handle configuration changes.